plat/common/x86/lcpu.c

```diff
@@ -76,6 +77,9 @@ int lcpu_arch_init(struct lcpu *this_lcpu)
 
 	traps_lcpu_init(this_lcpu);
 
+	wrkgsbase((__uptr)this_lcpu);
```

Comment:
Native builds will end up having `gs_base == k_gs_base == lcpu`. If a bincompat application ends up modifying `gs_base`, ok fine no problem we will know. The reason I still initialize both is because the `clone` code `uk_syscall_regs_popall` introduced in #1175 has a `swapgs` and is used by both native and bincompat, since they have to be able coexist.